bindings/c/test/mako/mako.c

```diff
@@ -478,172 +426,224 @@ int run_transaction(FDBTransaction *transaction, mako_args_t *args,
   int randstrlen;
   int rangei;
 
-  /* transaction */
   clock_gettime(CLOCK_MONOTONIC, &timer_per_xact_start);
-  for (i = 0; i < MAX_OP; i++) {
 
+ retryTxn:
```

Comment:
I don't see where you reset the transaction here?

Reply:
My understanding is that it's automatically handled by `fdb_transaction_on_error()`.  @ajbeamon Am I correct?

Reply:
If `on_error` returns successfully, then the transaction will have been reset for you (this is a soft reset that preserves retry related fields). If it throws an error, then I believe that it doesn't get reset.

Reply:
Thanks, that's my understanding too.  So, for a retryable errors, we don't want to explicitly reset, but for non-retryable errors, we do want to reset.

Reply:
Ah I see - `ReadYourWritesTransaction` has a different semantic than `Transaction` which is why I was a bit confused